fraud/service.go

```diff
@@ -1,57 +1,99 @@
 package fraud
 
 import (
+	"bytes"
 	"context"
 	"encoding/hex"
 	"errors"
 	"fmt"
 	"sync"
-	"time"
 
 	"github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p-core/host"
 	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
-)
 
-const (
-	// fetchHeaderTimeout duration of GetByHeight request to fetch an ExtendedHeader.
-	fetchHeaderTimeout = time.Minute * 2
+	"github.com/celestiaorg/celestia-node/params"
 )
 
-// service is responsible for validating and propagating Fraud Proofs.
+// fraudRequests is amount of external requests that will be done in order to get fraud proofs from another peers.
+const fraudRequests = 5
+
+var fraudProtocolID = protocol.ID(fmt.Sprintf("/fraud/v0.0.1/%s", params.DefaultNetwork()))
+
+// ProofService is responsible for validating and propagating Fraud Proofs.
 // It implements the Service interface.
-type service struct {
+type ProofService struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+
 	topicsLk sync.RWMutex
 	topics   map[ProofType]*pubsub.Topic
 
 	storesLk sync.RWMutex
 	stores   map[ProofType]datastore.Datastore
 
 	pubsub *pubsub.PubSub
+	host   host.Host
 	getter headerFetcher
 	ds     datastore.Datastore
+
+	enabledSyncer bool
```

Comment:
```suggestion
	syncerEnabled bool
```

Reply:
why?

Reply:
this is just an english fix, "enabledSyncer" doesn't fit correctly

Reply:
+ 1